### PR TITLE
FileJournal: reduce a lock/unlock about aio_lock

### PR DESCRIPTION
**Pull request by yangruifeng**
Signed-off-by: Ruifeng Yang <yangruifeng.09209@h3c.com>

### CODE REVIEW

File: src/os/FileJournal.cc [hunk 1]
```diff
@@ -1482,18 +1482,18 @@ void FileJournal::write_finish_thread_entry()
...
-    {
-      Mutex::Locker locker(aio_lock);
-      if (aio_queue.empty()) {
-	if (aio_stop)
-	  break;
-	dout(20) << "write_finish_thread_entry sleeping" << dendl;
-	write_finish_cond.Wait(aio_lock);
-	continue;
-      }
+    if (aio_queue.empty()) {
```
Comment (XinzeChi) [inline src/os/FileJournal.cc]: I think we should get the lock before aio_queue.empty()?

Comment (XinzeChi) [inline src/os/FileJournal.cc]: I think we should get the lock before aio_queue.empty()?

Comment (yangruifeng) [inline src/os/FileJournal.cc]: old version:
unlock after check_aio_completion();
while(true)
lock before if (aio_queue.empty())

Comment (yangruifeng) [inline src/os/FileJournal.cc]: old version:
unlock after write_finish_cond.Wait(aio_lock);
while(true)
lock before if (aio_queue.empty())

unlock/lock is unnecessary also

**Comment by XinzeChi** [comment]
LGTM

**Comment by yangruifeng** [comment]
@XinzeChi yes, thanks!

**Comment by yangruifeng** [comment]
@XinzeChi modified

**Comment by XinzeChi** [comment]
LGTM

**Comment by xiaoxichen** [comment]
I think this is really some coding style (using objects to manage resources, in effective C++)
Do we really see some benefit with this change?

**Comment by yangruifeng** [comment]
i think this can reduce some context switch, i will test it and give a report.

**Comment by branch-predictor** [comment]
Can you prove that this gives reasonable performance increase, while maintaining reliability and stability under ANY workload?

**Comment by markhpc** [comment]
We definitely will want to see some performance or CPU usage data to show an improvement with this change.  Thanks!

**Comment by yuyuyu101** [comment]
It seemed could reduce a extra lock/unlock in some cases, and it should be hard to get the little performance difference..... I suggest we close now

**Comment by yangruifeng** [comment]
@yuyuyu101 it is difficult to measure, close.

**Pull request closed without merging**